src/statements.md

```diff
@@ -54,17 +54,36 @@ fn outer() {
 > **<sup>Syntax</sup>**\
 > _LetStatement_ :\
 > &nbsp;&nbsp; [_OuterAttribute_]<sup>\*</sup> `let` [_PatternNoTopAlt_]
->     ( `:` [_Type_] )<sup>?</sup> (`=` [_Expression_] )<sup>?</sup> `;`
+>     ( `:` [_Type_] )<sup>?</sup> ( | `=` [_Expression_] | `=` [_Expression_]
+>     `else` [_BlockExpression_]) `;`
 
-A *`let` statement* introduces a new set of [variables], given by an
-irrefutable [pattern]. The pattern is followed optionally by a type
-annotation and then optionally by an initializer expression. When no
-type annotation is given, the compiler will infer the type, or signal
+A *`let` statement* introduces a new set of [variables], given by a [pattern].
+The pattern is followed optionally by a type annotation and then either ends,
+or is followed by an initializer expression plus an optional `else` block.
+When no type annotation is given, the compiler will infer the type, or signal
 an error if insufficient type information is available for definite
 inference. Any variables introduced by a variable declaration are visible
 from the point of declaration until the end of the enclosing block scope,
 except when they are shadowed by another variable declaration.
 
+If an `else` block is not present, the pattern has to be irrefutable.
+If an `else` block is present, the pattern may be refutable, and if it is, it
+will be checked on run time. On a mismatch, the `else` block is executed.
```

Comment:
I feel like specifying "run time" here may be over-specifying the behavior. For example, a `let` in a `const` will be evaluated at compile-time. It also isn't specifically stated what it means "to check", which isn't a verb typically used in the reference with pattern matching. These seem like two clauses to me, so I think it would be fine to state them separately, such as:

```suggestion
If an `else` block is present, the pattern may be refutable.
If the pattern does not match the initializer expression, the `else` block is executed.
```

Reply:
const-evaluation still has a "run-time" and compile time separation. But shrug.